### PR TITLE
feat: #8493, plugin helper for standardised link/button injection

### PR DESCRIPTION
**Pull request by samt**

### CODE REVIEW

**Comment by julianlam** [comment]
Okay, let's streamline the existing application of the `buildLinks` method here:

1. Remove `profile_menu` from `getAllData` method
1. In your call to `buildLinks`, instead of passing in `results.profile_menu.links`, pass in the links provided by core, in `getProfileMenu`, directly.
1. Remove `getProfileMenu`
1. Preserve the existing call to `filter:user.profileMenu`, however, deprecate it by adding it to `Plugins.deprecatedHooks` in `src/plugins/hooks.js` (the key is the old hook name, the value is the new hook name)

**Comment by julianlam** [comment]
Hey @samt just bumping to remind you of the changes needed before merging

**Comment by samt** [comment]
I've squashed this into a single, atomic commit, please let me know if any more changes are needed here 😃

File: src/plugins/hooks.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const utils = require('../utils');
 
 module.exports = function (Plugins) {
 	Plugins.deprecatedHooks = {
-
+		'filter:user.profileMenu': 'filter:accounts.profile',
```
Comment (julianlam) [inline src/plugins/hooks.js:9]: One last thing then it's good to merge... this hook name is too vague, we need a hook name that is indicative that it is for building out buttons. Ping me on slack.

Comment (julianlam) [reply src/plugins/hooks.js:9]: `filter:buildLinks.accounts/profile` please

**Pull request merged**